src/app/shared/components/post/post.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { IPost } from 'shared';
+
+@Component({
+  selector: 'app-post',
+  templateUrl: './post.component.html',
+  styleUrls: ['./post.component.scss'],
+})
+export class PostComponent implements OnInit {
+  @Input('post') post: IPost | null = null;
+
+  @Output() onLikeClick = new EventEmitter<string>();
+
+  isOpened: boolean = false;
+
+  constructor() {}
+
+  ngOnInit(): void {}
+
+  toggleOpened(event: MouseEvent): void {
+    event.stopPropagation();
+    this.isOpened ? (this.isOpened = false) : (this.isOpened = true);
+  }
+
+  get postLikes(): string {
+    if (this.post?.likes.length) {
+      return this.post.likes.length > 0 ? 'warn' : '';
+    }
+    return '';
+  }
+
+  addPostLike(event: MouseEvent) {
+    event.stopPropagation();
```

Comment:
why you stop propagation for each click?

Reply:
When I click on card with post, i navigate to page with single post. I need stop propagation to not trigger navigation, and add like to post.